pkg/apiserver/apiserver.go

```diff
@@ -80,19 +83,27 @@ func (c *Config) Complete() completedConfig {
 
 // New returns a new instance of CustomMetricsAdapterServer from the given config.
 // name is used to differentiate for logging.
-func (c completedConfig) New(name string, cmProvider provider.CustomMetricsProvider) (*CustomMetricsAdapterServer, error) {
+func (c completedConfig) New(name string, customMetricsProvider provider.CustomMetricsProvider, externalMetricsProvider provider.ExternalMetricsProvider, enableCustomMetricsAPI bool, enableExternalMetricsAPI bool) (*CustomMetricsAdapterServer, error) {
```

Comment:
no need for the extra arguments to enable APIs.  If a given provider is nil, don't instantiate it.  Otherwise, do.

Reply:
Done. I think it's still useful to have flags, so not removing them.